tests/unit/LSYiiValidatorsTest.php

```diff
@@ -27,4 +27,155 @@ public function testHtmlEntityDecodeFilter()
+     * Testing that the xssfilter attribute varies
+     * for regular users depending on filterxsshtml.
+     */
+    public function testXssFilterAttributeForRegularUsers()
+    {
+        //Create user.
+        $newPassword = createPassword();
+        $userId = \User::insertUser('test_user', $newPassword, 'John Doe', 1, 'jd@mail.com');
+
+        //Mocking regular user login.
+        \Yii::app()->session['loginID'] = $userId;
+        $regularUserValidator = new \LSYii_Validators();
+
+        $this->assertTrue(\Yii::app()->getConfig('filterxsshtml'), 'filterxsshtml should be true by default.');
+        $this->assertTrue($regularUserValidator->xssfilter, 'The xssfilter attribute should be true for regular users.');
+
+        //Changing filterxsshtml.
+        \Yii::app()->setConfig('filterxsshtml', false);
+        $newRegularUserValidator = new \LSYii_Validators();
+
+        $this->assertFalse(\Yii::app()->getConfig('filterxsshtml'), 'filterxsshtml was just changed to false.');
+        $this->assertFalse($newRegularUserValidator->xssfilter, 'The xssfilter attribute should be false for regular users with filterxsshtml set to false.');
+
+        //Returning to original values.
+        \Yii::app()->setConfig('filterxsshtml', true);
+        \Yii::app()->session['loginID'] = null;
+
+        //Delete user.
+        $user = \User::model()->findByPk($userId);
+        $user->delete();
+    }
+
+    /**
+     * Testing that any script or dangerous HTML is removed.
+     */
+    public function testXssFilterApplied()
+    {
+        $validator = new \LSYii_Validators();
+
+        $cases = array(
+            array(
+                'string'   => '<script>alert(`Test`)</script>',
+                'expected' => ''
+            ),
+            array(
+                'string'   => `{join('html_entity_decode("', '<script>alert("Test")</script>")')}`,
+                'expected' => ''
+            ),
+            array(
+                'string'   => '<title>html_entity_decode("<script>alert("Test")</script>")</title>',
+                'expected' => 'html_entity_decode("")'
+            ),
+            array(
+                'string'   => `{join('html_entity_decode("', '<s', 'cript>alert("Test")</script>")')}`,
+                'expected' => ''
+            ),
+            array(
+                'string'   => `{join('html_entity_decode("', '<', 'script>alert("Test")<', '/script>")')`,
+                'expected' => ''
+            ),
+            array(
+                'string'   => '<title>html_entity_decode("<script>alert("Test")</script>123456")</title>',
+                'expected' => 'html_entity_decode("123456")'
+            )
+        );
+
+        foreach ($cases as $case) {
+            $this->assertSame($case['expected'], $validator->xssFilter($case['string']), 'Unexpected filtered dangerous string.');
+        }
+    }
+
+    /**
+     * Testing that safe HTML tags are not removed.
+     */
+    public function testSafeHtml()
```

Comment:
Adding some testUnsageHtml ?

`<strong>strong` `<style>` <img onerror>`

?

Reply:
> Isn't that already tested on testXssFilterApplied?

No, testXssFilterApplied seems to test only script, not broken HTML.

But it's more new test to do. Can be added or not. Task is unclear here.

Reply:
Created new ticket: 18905: Create automatic test for XSS Purifier: Broken HTML